docs/assessment.md

```diff
@@ -224,6 +224,109 @@ When calculating a student's grade for a group assessment, PrairieLearn will alw
+      "canAssignRolesAtStart": true
+    },
+    {
+      "name": "Recorder",
+      "minimum": 1,
+      "maximum": 1
+    },
+    {
+      "name": "Reflector",
+      "minimum": 1,
+      "maximum": 1
+    },
+    {
+      "name": "Contributor"
+    }
+  ]
+}
+```
+
+| Attribute               | Type    | Default | Description                                                              |
+| ----------------------- | ------- | ------- | ------------------------------------------------------------------------ |
+| `name`                  | string  | -       | The name of the role.                                                    |
+| `minimum`               | integer | 0       | The minimum required number of this role in the assessment.              |
+| `maximum`               | integer | -       | The maximum required number of this role in the assessment.              |
+| `canAssignRolesAtStart` | boolean | false   | Allow students with this role to assign roles throughout the assessment. |
+
+Students select their roles before starting an assessment, but they can change their roles mid-assessment if needed. As a safeguard against invalid role configurations, PrairieLearn prevents all question submissions if a group's role configuration does not meet the instructor's specification.
+
+### Adding permissions for an assessment
+
+Permissions can be configured at the _assessment_, _zone_, or _question_ level.
+
+The schema for permissions is defined as follows:
+
+```json
+{
+  "canView": ["Manager", "Reflector", "Recorder", "Contributor"],
+  "canSubmit": ["Recorder"]
+}
+```
+
+| Attribute   | Type            | Default | Description                                                     |
+| ----------- | --------------- | ------- | --------------------------------------------------------------- |
+| `canView`   | Array of string | -       | The names of roles that can view this part of the assessment.   |
+| `canSubmit` | Array of string | -       | The names of roles that can submit this part of the assessment. |
+
+Setting either attribute to `[]` (empty array) means that **no role** can view/submit that part. Setting either attribute to `null` means that **every role** can view/submit that part.
+
+Permissions defined at a higher level are propagated down the assessment hierarchy (assessment -> zone -> question), but permissions defined at lower levels will override those from the higher level. For example, if the permissions in the example above are configured for the assessment, but a question's permissions are configured to `canSubmit: ["Reflector"]`, then only the _Reflector_ can submit that particular question.
+
+### Students joining assessments with custom group roles
+
+When students join a group, they are automatically assigned a role. Students can always view the roles of other students in the group, both before and during an assessment. More information about the assessment's group roles can be expanded in the dropdown menu.
```

Comment:
This is not a dropdown menu in the usual sense - maybe say something like this instead?

```suggestion
When students join a group, they are automatically assigned a role. Students can always view the roles of other students in the group, both before and during an assessment. Students can click "View role info" to see more information about the assessment's group roles.
```

You'll need to change the usage of "dropdown menu" a few lines down as well.